lib/krane/kubernetes_resource/stateful_set.rb

```diff
@@ -19,6 +19,9 @@ def status
     end
 
     def deploy_succeeded?
+      success = observed_generation == current_generation &&
+        desired_replicas == status_data['readyReplicas'].to_i &&
+        status_data['currentRevision'] == status_data['updateRevision']
```

Comment:
This check is currently broken in upstream Kubernetes for `OnDelete`, but if we remove the check the method will return immediately, before the pods are actually restarted successfully.

We could (re)implement some of the controller logic to check for individual pods but I believe running with `--no-verify-result` until the upstream fix is the better option.

Upstream fix: https://github.com/kubernetes/kubernetes/pull/106059